platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/binding/api/impl/ValidatorImpl.java

```diff
@@ -96,10 +96,14 @@ public void validateAuthnRequest(
   }
 
   @Override
-  public void validateRelayState(String relayState) {
+  public void validateRelayState(String relayState, boolean strictRelayState) {
     LOGGER.debug("Validating RelayState");
     if (relayState != null && relayState.length() > 80) {
-      LOGGER.warn("RelayState has invalid size: {}", relayState.length());
+      if (strictRelayState) {
+        throw new IllegalArgumentException("RelayState must be 80 bytes or shorter");
+      } else {
+        LOGGER.warn("RelayState has invalid size: {}", relayState.length());
```

Comment:
Added the warning log regardless of fail or not.

As for the user perspective: I don't know what that actually looks like. A good guess is that the browser will just display the default `500 error` page, since we don't provide our own.